scripts/posteffects/posteffect-ssao.js

```diff
@@ -0,0 +1,568 @@
+            "uniform int uBilatSampleCount;",
+            "uniform float uFarPlaneOverEdgeDistance;",
+            "uniform float uBrightness;",
+            "",
+            "float random(const highp vec2 w) {",
+            "    const vec3 m = vec3(0.06711056, 0.00583715, 52.9829189);",
+            "    return fract(m.z * fract(dot(w, m.xy)));",
+            "}",
+            "",
+            "float bilateralWeight(in float depth, in float sampleDepth) {",
+            "    float diff = (sampleDepth - depth) * uFarPlaneOverEdgeDistance;",
+            "    return max(0.0, 1.0 - diff * diff);",
+            "}",
+            "",
+            "void tap(inout float sum, inout float totalWeight, float weight, float depth, vec2 position) {",
+            "    // ambient occlusion sample",
+            "    float ssao = texture2D( uSSAOBuffer, position ).r;",
+            "    float tdepth = -getLinearScreenDepth( position );",
+            "",
+            "    // bilateral sample",
+            "    float bilateral = bilateralWeight(depth, tdepth);",
+
+            "    bilateral *= weight;",
+            "    sum += ssao * bilateral;",
+            "    totalWeight += bilateral;",
+            "}",
+            "",
+            "void main() {",
+            "    highp vec2 uv = vUv0; // variable_vertex.xy; // interpolated at pixel's center",
+            "",
+            "    // we handle the center pixel separately because it doesn't participate in bilateral filtering",
+            "    float depth = -getLinearScreenDepth(vUv0); // unpack(data.gb);",
+            "    float totalWeight = 0.0; // float(uBilatSampleCount*2+1)*float(uBilatSampleCount*2+1);",
+            "    float ssao = texture2D( uSSAOBuffer, vUv0 ).r;",
+            "    float sum = ssao * totalWeight;",
+            "",
+            "    for (int x = -uBilatSampleCount; x <= uBilatSampleCount; x++) {",
+            "       for (int y = -uBilatSampleCount; y < uBilatSampleCount; y++) {",
+            "           float weight = 1.0;",
+            "           vec2 offset = vec2(x,y)*uResolution.zw;",
+            "           tap(sum, totalWeight, weight, depth, uv + offset);",
+            "       }",
+            "    }",
+            "",
+            "    float ao = sum / totalWeight;",
+            "",
+            "    // simple dithering helps a lot (assumes 8 bits target)",
+            "    // this is most useful with high quality/large blurs",
+            "    // ao += ((random(gl_FragCoord.xy) - 0.5) / 255.0);",
+            "",
+            "    vec4 inCol = texture2D( uColorBuffer,  vUv0 );",
+            "",
+            "    ao = mix(ao, 1.0, uBrightness);",
+            "    gl_FragColor.rgb = inCol.rgb * ao;",
+            "    gl_FragColor.a = 1.0;",
+            "}"
+        ].join("\n")
+    });
+
+    // Render targets
+    var width = graphicsDevice.width;
+    var height = graphicsDevice.height;
+    this.targets = [];
+    for (var i = 0; i < 1; i++) {
```

Comment:
Why is there a `for` loop here if there's only 1 iteration?

Reply:
copy & paste from bloom, but changing 2 to 1 iterations. I'll clean this up.